src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java

```diff
@@ -135,9 +136,51 @@ private Object buildArg(Type p, GraphQLType graphQLType, Object arg) {
         }
     }
 
-    private Object getFieldValue(Object source, String fieldName) throws IllegalAccessException, NoSuchFieldException {
-        Field field = source.getClass().getDeclaredField(fieldName);
-        field.setAccessible(true);
-        return field.get(source);
+    private Object getGraphQLFieldValue(Object source, String fieldName) throws IllegalAccessException, NoSuchFieldException, InvocationTargetException {
+        Method method = getMethod(source.getClass(), fieldName, "");
+        Method getMethod = getMethod(source.getClass(), fieldName, "get");
+        Method isMethod = getMethod(source.getClass(), fieldName, "is");
+        if (method != null) {
+            return method.invoke(source);
+        } else if (getMethod != null) {
+            return getMethod.invoke(source);
+        } else if (isMethod != null) {
+            return isMethod.invoke(source);
+        } else {
+            Field field = getField(source.getClass(), fieldName);
+            if (field != null) {
+                field.setAccessible(true);
+                return field.get(source);
+            } else {
+                throw new NoSuchFieldException("No GraphQL field found");
+            }
+        }
+    }
+
+    private Method getMethod(Class<?> clazz, String name, String prefix) {
+        String prefixedName = createPrefix(prefix, name);
+        Method method = null;
+        while (clazz != null && method == null) {
+            try {
+                method = clazz.getDeclaredMethod(prefixedName);
+            } catch (Exception e) {
+            }
+            clazz = clazz.getSuperclass();
+        }
+
+        return method;
     }
+
+    private Field getField(Class<?> clazz, String name) {
+        Field field = null;
+        while (clazz != null && field == null) {
+            try {
+                field = clazz.getDeclaredField(name);
```

Comment:
should it be "getDeclaredField" or "getField"?

Reply:
getField returns only public fields (but for entire hirearchy), but we need private ones as well. So this method runs over the entire hirearchy and searches for private methods with this name.